src/insight/metrics/base.py

```diff
@@ -131,10 +219,40 @@ class DataFrameMetric(_Metric):
         >>> metric(df)
         3
     """
+    def __call__(self,
+                 df: pd.DataFrame,
+                 dataset_name: str = None,
+                 session: Session = None,
+                 dataset_rows: int = None,
+                 dataset_cols: int = None,
```

Comment:
These could be calculated from `df` instead of being additional arguments - it might make the function a bit more straight forward.